src/behaviours/master-data.js

```diff
@@ -0,0 +1,82 @@
+import React, {Component, PropTypes} from 'react';
+import {loadMasterData} from '../actions/master-data';
+import {pick} from 'lodash/object';
+import {capitalize} from 'lodash/string';
+
+const MASTER_DATA_PROPTYPE = PropTypes.arrayOf(
+    PropTypes.shape({
+      name: PropTypes.string.isRequired,
+      service: PropTypes.func.isRequired,
+      cacheDuration: PropTypes.number
+})).isRequired;
+
+const MASTER_DATA_CONTEXT_TYPE = {
+  masterDataLoaders: MASTER_DATA_PROPTYPE
+}
+
+
+// Connect the component to the load method of the master data.
+// You have to pass an array of names in which you are interested.
+// These names should correspond to the configuration you provide to the `MasterDataProvider`.
+// It add as props to the connected component the loaders for each requested master data with the name `loadMasterDataName`
+// Where name is what you give to the names property of the connector.
+// Example Call
+//```jsx
+//const MyComponentWhichUsesAMasterDataLoader = (props) => <div><button = >LoadMyList</button>{props.name}</div>;
+//const MyMasterDataConnectedComponent =  connectMasterData(['sandwichTypes'])(MyComponentWhichUsesAMasterDataLoader)
+//```
+const connectMasterData = (names = []) => {
+  return function connectComponent(ComponentToConnect) {
+      const MasterDataConnectedComponent = (props, {masterDataLoaders}) => {
+          const wantedMasterDataLoaders = masterDataLoaders.reduce((res, current) => {
+              if(names.indexOf(current.name) !== -1){
+                res[`loadMasterData${capitalize(current.name)}`] = () => loadMasterData(current.name, current.service, current.cacheDuration);
+              }
+              return res;
+          }, {});
+          if(Object.keys(wantedMasterDataLoaders) === 0){
+            console.warn(`connectMasterData: your keys ${Object.keys(names)} are not in the masterDataLoaders ${masterDataLoaders.reduce((res, current) => [...res, current.name]), [])}`)
+          }
+          const {_behaviours, ...otherProps} = props;
+          const behaviours = {isConnectedMasterData: true, ..._behaviours}
```

Comment:
I don't get your reduce, wouldn't it be : `loadAllMasterData = () => Object.keys(wantedMasterDataLoaders).reduce((res, current) => wantedMasterDataLoaders[current]());`